hw02-ArrayList/src/main/java/Main.java

```diff
@@ -0,0 +1,52 @@
+import java.util.*;
+
+public class Main {
+    public static void main(String[] args) {
+        System.out.println("--- Создание объектов ---");
+
+        DIYArrayList<Integer> diyArrayList1 = new DIYArrayList<>();
+        System.out.println("1 конструктор: " + diyArrayList1.toString());
+
+        DIYArrayList<Integer> diyArrayList2 = new DIYArrayList<>(30);
+        System.out.println("2 конструктор: " + diyArrayList2.toString());
+
+        DIYArrayList<String> diyArrayList3 = new DIYArrayList<>(Arrays.asList("Строка 1",
+                                                                              "Строка 2",
+                                                                              "Строка 3",
+                                                                              "Строка 4",
+                                                                              "Строка 5",
+                                                                              "Строка 6",
+                                                                              "Строка 7",
+                                                                              "Строка 8",
+                                                                              "Строка 9"));
+        System.out.println("3 конструктор: " + diyArrayList3.toString());
+
+
+        System.out.println("\r\n\r\n\r\n--- тестируем Collections.addAll(Collection<? super T> c, T... elements) ---");
+        System.out.println("Добаление списка элементов {\"Новая строка 1\", \"Новая строка 2\", \"Новая строка 3\", \"Новая строка 4\", \"Новая строка 5\"}");
+        Collections.addAll(diyArrayList3, "Новая строка 1", "Новая строка 2", "Новая строка 3", "Новая строка 4", "Новая строка 5");
+        System.out.println("После метода addAll: " + diyArrayList3.toString());
+
+
+        System.out.println("\r\n\r\n\r\n--- тестируем Collections.static <T> void copy(List<? super T> dest, List<? extends T> src) ---");
+        DIYArrayList<String> diyArrayDest = new DIYArrayList<>(Arrays.asList("1", "2", "3"));
+        System.out.println("DIYArrayList(dest) " + diyArrayDest.toString());
+        System.out.println("DIYArrayList(source) " + diyArrayList3.toString());
+        Collections.copy(diyArrayList3, diyArrayDest);
```

Comment:
У вас перепутаны местами src и dest
```suggestion
        Collections.copy(diyArrayDest, diyArrayList3);
```

```java
public static <T> void copy(List<? super T> dest, List<? extends T> src) {
```

Проверьте работу на коллекции с больше, чем 20 элементов (сейчас у вас так и есть, не уменьшайте размер коллекции).